contracts/protocol/clients/FermionFNFTPriceManager.sol

```diff
@@ -238,43 +255,48 @@ contract FermionFNFTPriceManager is FermionErrors, ERC2771Context, IFermionFNFTP
      * - `AlreadyVoted` if the caller has already voted and has no additional fractions to contribute.
      *
      * @param _voteYes True to vote YES, false to vote NO.
+     * @param _proposalId The ID of the proposal to vote on.
      */
-    function voteOnProposal(bool _voteYes) external {
+    function voteOnProposal(bool _voteYes, uint256 _proposalId) external {
         FermionTypes.FermionFractionsStorage storage fractionStorage = Common._getFermionFractionsStorage();
         uint256 currentEpoch = fractionStorage.currentEpoch;
-        FermionTypes.PriceUpdateProposal storage proposal = Common
-            ._getBuyoutAuctionStorage(currentEpoch)
-            .currentProposal;
-        address erc20Clone = fractionStorage.epochToClone[currentEpoch];
+        FermionTypes.BuyoutAuctionStorage storage $ = Common._getBuyoutAuctionStorage(currentEpoch);
+        if (_proposalId >= $.priceUpdateProposals.length) revert FractionalisationErrors.InvalidProposalId();
+        FermionTypes.PriceUpdateProposal storage proposal = $.priceUpdateProposals[_proposalId];
         address msgSender = _msgSender();
+        FermionTypes.PriceUpdateVoter memory voter = $.voters[msgSender];
+
+        address erc20Clone = fractionStorage.epochToClone[currentEpoch];
         uint256 fractionsBalance = IERC20(erc20Clone).balanceOf(msgSender);
 
-        if (proposal.state != FermionTypes.PriceUpdateProposalState.Active) {
+        if (proposal.state != FermionTypes.PriceUpdateProposalState.Active)
             revert FractionalisationErrors.ProposalNotActive(proposal.proposalId);
```

Comment:
```suggestion
            revert FractionalisationErrors.ProposalNotActive(_proposalId);
```
`proposal.proposalId` should be the same as `_proposalId` which is more gas efficient. The are few other places where it can be replaced.